js/src/forum/extend.ts

```diff
@@ -0,0 +1,9 @@
+import Extend from 'flarum/common/extenders';
+import Discussion from 'flarum/common/models/Discussion';
+
+export default [
+  new Extend.Model(Discussion) //
+    .attribute<boolean>('frontpage')
+    .attribute<string>('frontdate')
```

Comment:
Should this not be `Date|null`? 🤔

Reply:
I was thinking so as well, but the backend returns a String

Reply:
Perhaps we should cast this on the backend model in that case, or might that be a breaking change I wonder. Will check shortly

Reply:
Actually, I'm not so sure anymore where exactly I saw that it's returning a string. I can take a look at it again if you want

Reply:
@imorland I had another look, we are casting it as `datetime` in the backend, but PHP returns a string in the ISO 8601 format so it's a `string` in the frontend. We could convert it to a JS Date, but I'm not sure if that's something we want to do?

Reply:
Yep, makes sense.. changed in https://github.com/FriendsOfFlarum/frontpage/pull/32/commits/f65e9743d7b02703ac5bc776953fdd5bf13ef7f1